code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -0,0 +1,421 @@
+		return 1
+
+/obj/item/modular_computer/examine(mob/user)
+	..()
+	if(damage > broken_damage)
+		user << "<span class='danger'>It is heavily damaged!</span>"
+	else if(damage)
+		user << "<span class='warning'>It is damaged.</span>"
+
+/obj/item/modular_computer/update_icon()
+	cut_overlays()
+	if(!enabled)
+		icon_state = icon_state_unpowered
+	else
+		icon_state = icon_state_powered
+		if(active_program)
+			add_overlay(active_program.program_icon_state ? active_program.program_icon_state : icon_state_menu)
+		else
+			add_overlay(icon_state_menu)
+
+	if(damage > broken_damage)
+		add_overlay("bsod")
+		add_overlay("broken")
+
+
+// On-click handling. Turns on the computer if it's off and opens the GUI.
+/obj/item/modular_computer/attack_self(mob/user)
+	if(enabled)
+		ui_interact(user)
+	else
+		turn_on(user)
+
+/obj/item/modular_computer/proc/turn_on(mob/user)
+	var/issynth = issilicon(user) // Robots and AIs get different activation messages.
+	if(damage > broken_damage)
+		if(issynth)
+			user << "<span class='warning'>You send an activation signal to \the [src], but it responds with an error code. It must be damaged.</span>"
+		else
+			user << "<span class='warning'>You press the power button, but the computer fails to boot up, displaying variety of errors before shutting down again.</span>"
+		return
+
+	// If we have a recharger, enable it automatically. Lets computer without a battery work.
+	if(recharger)
+		recharger.enabled = 1
+
+	if(processor_unit && use_power()) // use_power() checks if the PC is powered
+		if(issynth)
+			user << "<span class='notice'>You send an activation signal to \the [src], turning it on.</span>"
+		else
+			user << "<span class='notice'>You press the power button and start up \the [src].</span>"
+		enabled = 1
+		update_icon()
+		ui_interact(user)
+	else // Unpowered
+		if(issynth)
+			user << "<span class='warning'>You send an activation signal to \the [src] but it does not respond.</span>"
+		else
+			user << "<span class='warning'>You press the power button but \the [src] does not respond.</span>"
+
+// Process currently calls handle_power(), may be expanded in future if more things are added.
+/obj/item/modular_computer/process()
+	if(!enabled) // The computer is turned off
+		last_power_usage = 0
+		return 0
+
+	if(damage > broken_damage)
+		shutdown_computer()
+		return 0
+
+	if(active_program && active_program.requires_ntnet && !get_ntnet_status(active_program.requires_ntnet_feature))
+		active_program.event_networkfailure(0) // Active program requires NTNet to run but we've just lost connection. Crash.
+
+	for(var/I in idle_threads)
+		var/datum/computer_file/program/P = I
+		if(P.requires_ntnet && !get_ntnet_status(P.requires_ntnet_feature))
+			P.event_networkfailure(1)
+
+	if(active_program)
+		if(active_program.program_state != PROGRAM_STATE_KILLED)
+			active_program.process_tick()
+			active_program.ntnet_status = get_ntnet_status()
+			active_program.emagged = emagged
```

Comment:
So the programs that were downloaded after emagging would get emaged flag too. Still not sure about this, will investigate.